tests/main/confinement-classic/task.yaml

```diff
@@ -5,13 +5,19 @@ details: |
     this test (classic and jailmode) and the snap (this particular one) should
     function correctly in both cases.
 systems: [-ubuntu-core-16-*]
-environment:
-    INSTALL_FLAGS/classic: --classic
-    INSTALL_FLAGS/classic_and_jailmode: --classic --jailmode
 prepare: |
-    make -C test-snapd-hello-classic clean
-    make -C test-snapd-hello-classic
-    snap install $INSTALL_FLAGS --dangerous ./test-snapd-hello-classic/test-snapd-hello-classic_0.1_*.snap
+    run_install() {
```

Comment:
I don't get one thing. The `environment` section above spawns two instances of this test, each with its own prepare code. Now you roll this into one where we `run_install --classic` and `run_install --jailmode` in one prepare step. This effectively means that we no longer test `--classic` sans `--jailmode`. Am I missing anything?